libbeat/reader/readfile/limit.go

```diff
@@ -37,7 +39,10 @@ func NewLimitReader(r reader.Reader, maxBytes int) *LimitReader {
 func (r *LimitReader) Next() (reader.Message, error) {
 	message, err := r.reader.Next()
 	if len(message.Content) > r.maxBytes {
-		message.Content = message.Content[:r.maxBytes]
+		n = copy(message.Content, message.Content[:r.maxBytes])
```

Comment:
`copy` doesn't allocate or change the size of the underlying array.

it must say:

```
tmp := make([]byte, len(message.Content))
copy(tmp, message.Content)
message.Content = tmp
```